### ISSUE TITLE
Raise error when an unknown or invalid `strict_max_version` is specified in manifest

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#14654]
In https://github.com/mozilla/addons-server/issues/10864#issuecomment-470956849 we decided to ignore invalid `strict_max_version` and use `*` instead but this is causing more bad than good. Instead of doing this, our plan is to [add a specific API for release management to use to create the relevant `AppVersion`](https://github.com/mozilla/addons-server/issues/14649) and stop silently rewriting invalid `strict_max_version`.

**Comment by diox** [comment]
Waiting on https://bugzilla.mozilla.org/show_bug.cgi?id=1663453 to be verified fixed. It's close - it's been tested on stage successfully.

**Comment by AlexandraMoga** [comment]
I've verified this on stage by uploading several add-ons (extensions and language packs) with different `strict_min_version` and `strict_max_version` combinations.

For the `min` version, we are still raising the errors we raised before for the lowest supported version:
![image](https://user-images.githubusercontent.com/31961530/94688815-c8634b80-0336-11eb-8d85-140cbf269437.png)


For the `max` version, if the version is not defined in https://addons.allizom.org/en-US/firefox/pages/appversions/, a validation error is raised:
![image](https://user-images.githubusercontent.com/31961530/94688939-ea5cce00-0336-11eb-9c08-f63489412a14.png)


We are still also raising an error if the version string is invalid.

@diox there's one potential issue with versions created for Firefox but missing in Firefox for Android. For example, if versions `86.0 - 86.*` are available only for Firefox, an extension that uses this as `strict_max_version` won't pass validation, even though Firefox for Android is not marked as supported during upload:

![image](https://user-images.githubusercontent.com/31961530/94688994-fb0d4400-0336-11eb-94ae-903b469dbb50.png)


This error will not be raised for language packs though, probably because they ignore Firefox for Android by default?

**Comment by diox** [comment]
ugh that's kinda annoying. Hard to fix because the manifest doesn't know about Firefox vs Firefox for Android... I normally would say we don't care, but I noticed that https://bugzilla.mozilla.org/show_bug.cgi?id=1663453 only cared about Firefox and not Firefox for Android so we need to do *something*. Re-opening to deal with that.

**Comment by diox** [comment]
@AlexandraMoga technically our validation doesn't care about applications, so I'm going to fix this differently, by creating versions for both Firefox and Firefox for Android in the API built in https://github.com/mozilla/addons-server/issues/14649

**Comment by AlexandraMoga** [comment]
@diox there seems to be an issue with this approach and how it handles `strict_min_version`. 
For example, I've created appversions `112.0 - 112.*` by submitting a new request to `/api/v4/applications/firefox/112.0/`. This creates an entry for `112.0 - 112.*` in [appversions](https://addons-dev.allizom.org/en-US/firefox/pages/appversions/) for both Firefox and Firefox for Android.
Then, I submitted an extension and a language pack with strict compatibility set to:
```
"strict_min_version": "112.0"
"strict_max_version": "112.*"
```
The DevHub manage versions page will show the following compatibility range for both the extension and the language pack:

![image](https://user-images.githubusercontent.com/31961530/95206438-5f288000-07ef-11eb-925c-9f140527d706.png)

[Lang pack example](https://addons-dev.allizom.org/en-US/developers/addon/release-lang-pack777/versions/1692139)
[Extension example](https://addons-dev.allizom.org/en-US/developers/addon/min_max_appversion4/versions/1692138)

I did not reproduce the issue when I used an appversion created before this new patch landed.
I think this will create some problems with language pack submissions especially. Let me know if you want a new issue filed or if you prefer to just reopen this issue.

**Comment by diox** [comment]
Woah, this is a good find. It took me some time to figure out what's happening there... It's not tied to my changes at all, it's because of the version you used - we end up considering "112" is lower than "42.0" so we pick that instead. Please open a new issue for this.

**Comment by AlexandraMoga** [comment]
I've filed #15659

This issue can be marked as verified fixed now.
